AzurePowershell.Test.targets

```diff
@@ -56,20 +56,30 @@
 	</Target>
 
 	<Target Name="InvokeXUnit">
-		<Message Importance="high" Text="Running XUnit tests" />
-		<MakeDir Directories="$(TestOutputDirectory)" ContinueOnError="false" />
+		<PropertyGroup>
+			<TestName>$(XUnitTests.Split('\').Length)</TestName>
+			<TestNamePathLength>$([MSBuild]::Subtract($(TestName), 1))</TestNamePathLength>
+			<TestSubFileName>$(XUnitTests.Split('\')[$(TestNamePathLength)])</TestSubFileName>
+			<TestFileName>$(TestOutputDirectory)\$(TestSubFileName).html</TestFileName>
+		</PropertyGroup>
+		<Message Importance="high" Text="Running XUnit tests... $(TestFileName)" />
 
 		<xunit
-			Assemblies="@(XUnitTests)"
+			Assemblies="$(XUnitTests)"
 			AppDomains="true"
 			ShadowCopy="false"
 			ParallelizeTestCollections="false"
 			ParallelizeAssemblies="false"
 			IncludeTraits="AcceptanceType=CheckIn"
-			Html="$(TestOutputDirectory)\AzurePowershellTestResults.html"
+			Html="$(TestFileName)"
 			DiagnosticMessages="false"
-			ContinueOnError="false"
-			Condition=" @(XUnitTests) != '' "/>
+			ContinueOnError="true"
+			Condition=" $(XUnitTests) != '' ">
+			<Output TaskParameter="ExitCode" ItemName="XUnitExitCode"/>
+		</xunit>
+
+		<Exec Command="&quot;$(PowerShellCommand)&quot; -NonInteractive -NoLogo -NoProfile -Command &quot;. $(LibraryToolsFolder)\CleanupTestFiles.ps1 -ExitCode %(XUnitExitCode.Identity) -FileName $(TestSubFileName) -FolderName $(TestOutputDirectory) &quot;"/>
```

Comment:
@maddieclayton since the `CleanupTestFiles.ps1` script is so small, we should be able to replace the script call with one line of PowerShell:

```powershell
Move-Item -Path $(TestOutputDirectory)/$(TestSubFileName).html -Destination $(TestOutputDirectory)/$(if(%(XUnitExitCode.Identity) -eq 0){"PassingTests"}else{"FailingTest"})/$(TestSubFileName).html
```